modules/test/playwright/tests/cookies-banner-web/cookiesBannerScript.spec.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {expect, mergeTests} from '@playwright/test';
+
+import {isolatedLayoutTest} from '../../fixtures/isolatedLayoutTest';
+import {loginTest} from '../../fixtures/loginTest';
+import {SystemSettingsPage} from '../../pages/configuration-admin-web/SystemSettingsPage';
+import {waitForSuccessAlert} from '../../utils/waitForSuccessAlert';
+
+export const test = mergeTests(isolatedLayoutTest(), loginTest());
+
+test('@LPD-25701 Cookie Banner Script', async ({layout, page}) => {
+	await test.step('Go to page and click edit', async () => {
+		await page.goto(layout.friendlyURL);
+
+		const editButton = page.getByRole('link', {name: 'Edit'});
+
+		await editButton.waitFor({state: 'visible'});
+		await editButton.click();
+	});
+
+	await test.step('Drag and drop html to page', async () => {
+		const searchFragment = page.getByLabel('Search Fragments and Widgets');
+
+		await searchFragment.waitFor({state: 'visible'});
+		await searchFragment.click();
+		await searchFragment.fill('html');
+
+		const htmlItem = page.getByRole('menuitem', {
+			name: 'HTML Add HTML Mark HTML as Favorite',
+		});
+
+		await htmlItem.waitFor({state: 'visible'});
+
+		await htmlItem.dragTo(page.locator('#page-editor div').nth(1));
+	});
+
+	await test.step('Add script to html and save page', async () => {
+		const htmlExample = page.getByText('HTML Example');
+
+		await htmlExample.waitFor({state: 'visible'});
+		await htmlExample.click();
+		await htmlExample.click();
+		await htmlExample.click();
+
+		const htmlSection = page.locator('.CodeMirror-scroll');
+
+		await htmlSection.waitFor({state: 'visible'});
+		await htmlSection.click();
+
+		const textarea = page.locator('textarea');
+
+		await textarea.press('Control+a');
+		await textarea.press('Backspace');
+		await textarea.fill(
+			'<h1 id="test">HTML Example</h1>\n' +
+				'\n' +
+				'<script type="text/plain" data-third-party-cookie="CONSENT_TYPE_FUNCTIONAL">\n' +
+				'      document.getElementById(\'test\').style.backgroundColor = "#ff0000"\n' +
+				'</script>'
+		);
+
+		await page
+			.getByRole('button', {
+				name: 'Save',
+			})
+			.click();
+
+		await page.getByLabel('Publish', {exact: true}).click();
+
+		await waitForSuccessAlert(
+			page,
+			`Success:The page was published successfully.`
+		);
+	});
+
+	await test.step('Enable Third Party Cookies', async () => {
+		const systemSettingsPage = new SystemSettingsPage(page);
```

Comment:
This will work fine, but a more cleaner pattern is to merge in `systemSettingsPageTest` fixture, and use the exposed `systemSettingsPage`, like [here](https://github.com/liferay/liferay-portal/blob/master/modules/test/playwright/tests/layout-set-prototype-web/siteTemplatesWithWebContent.spec.ts#L166). Some fixtures, like this one, just expose the page variable, so it'll work either way. But sometimes, fixures need to do some preparation magic, so it's a good practice to always use them.

Reply:
Oh nice, thank you!